### DISCUSSION TITLE
Type of the "OpenAPI Fundamentals" course

### DISCUSSION THREAD

**Discussion by karelhusa** [OAI/OAI-Courses#10] [General]
Is OpenAPI Fundamentals course going to be:
- On-line course with hands-on learning ("try it" exercises) for individuals
- A training as a in-person workshop
- Presentation for a conference (API Days-like)
- Any other?

**Comment by karelhusa** [comment]
The main question for me is the format, usage and goal of the training. Whether the goal a piece of new knowledge or new skills. If we focus on knowledge, the presentation style may be fine. If we want to build skills I would go for hands-on training.

As a trainer (and CS teacher) I always go for "hands-on" to raise the engagement of students. Also, people learn the most by doing something.  

The outcome of the first module (new skills) can be: 
- I can check an OpenAPI definition and find which operations the API provides.
- I can import the OpenAPI definition into Postman and call the API operations I need.
- I can check the structure of operation's request and response to see what data it provides.

**Reply by SensibleWood** [reply]
This course is intended to be several things:

* An accumulation of knowledge: The student should be able to start from scratch without necessarily knowing what an API specification language is or know anything about OpenAPI itself. They can start small, and work their way through to understand why OpenAPI exists, its structure, get into practical examples, and then go advanced in Module 4.
* A resource for others: The scripts are written the way they are as they idea is that **_anyone_** can pick up the materials and use them. That's why they are called scripts, and read as such. They are background material for a trainer to pick up and use in any scenario.
* The basis of an online course: This material will be grow - modules 3 and 4 are yet to be written - into a full course to be hosted on LF Training (https://training.linuxfoundation.org/). As you can see from the [overview](https://github.com/OAI/OAI-Courses/blob/main/src/openapi-v31-fundamentals/README.md) these later modules will be much more practical.

So to get to the nub of your question: This is supposed to be presentation style to start with, and move into hands-on as the topics get more advanced. Feasibly someone would be able to rock up and take module 1, and walk away knowing enough about OpenAPI to do something with it. They could go right through and do the lot, and be able to apply their knowledge and new found skill practically.

One of the goals of this thing is to allow a whole of range of people to take as little or as much as they want. There are parts of this course where doing something hands-on just isn't required. For example, a  module 1 student might be a product manager with no interest in sending a Postman request. They just want the knowledge of why the phrase "OpenAPI" appears in their conversations with developers.

**Reply by karelhusa** [reply]
@SensibleWood Thanks for explanation, starting with a wiki-style knowledge a taking specific topics for training modules seems a good strategy.
